src/DependencyInjection/Configuration.php

```diff
@@ -91,12 +91,18 @@ private function getDynamicTemplateNode()
                         ->scalarNode('path_unmatch')->end()
                         ->scalarNode('match_pattern')->end()
                         ->arrayNode('mapping')
-                            ->prototype('variable')
+                            ->variablePrototype()
                                 ->treatNullLike([])
                             ->end()
                         ->end()
                     ->end()
                 ->end()
+                ->validate()
+                    ->ifTrue(static function ($v) {
+                        return 1 !== \count($v);
+                    })
+                    ->thenInvalid('Dynamic template should consist of a single named object: %s.')
```

Comment:
Using `named object` wording as stated in the doc: https://www.elastic.co/guide/en/elasticsearch/reference/7.x/dynamic-templates.html#dynamic-templates